### PR TITLE
Fix for issue of sidebar not closing on mobile.

### PR DESCRIPTION
**Pull request by bjthompson805**
Fix for issue #66 to close sidebar if an item is clicked that was already previously selected.

### CODE REVIEW

File: src_js/components/sidebar/TableOfContents.tsx [hunk 1]
```diff
@@ -40,6 +40,13 @@ export default function TableOfContents(props: PropsType) {
     };
 
     window.addEventListener('hashchange', hash_listener, false);
+
+    // Also account for user clicking the same sidebar item that is already selected [Fixes issue #66].
+    var sidebarTocListElements : any = document.body.querySelector('aside nav ul.primer-spec-toc-list');
```
Comment (seshrs) [inline src_js/components/sidebar/TableOfContents.tsx:45]: If we're going to add a click-handler, we might as well add it on the entire `<nav>` — in this case, we don't even *need* to use `useEffect()`! Move this logic to an `onClick` attribute of the `<nav>` in the `render()` method.

```jsx
<nav
  id="primer-spec-toc"
  onClick={() => {
    // Move the logic here
  }}
>
  {tocNodes}
</nav>
```

**Review by seshrs — CHANGES_REQUESTED** [review]
Thanks for the PR, this is awesome! I like your idea of adding a click-handler on the entire table of contents.

One side-effect is that if a user clicks anywhere inside the Table of Contents (even if it's not a link), the Sidebar will close on mobile. However, I think that's a reasonable behavior 😄

**Comment by bjthompson805** [comment]
On this one, I had intended to add it only to the links. I think if the user clicks the sidebar it shouldn't close it until they click a link. What I could do is add it to the <a> tags in the sidebar. What do you think?

**Comment by seshrs** [comment]
> I think if the user clicks the sidebar it shouldn't close it until they click a link.

Ah I see. I held the same opinion when I first created issue #66. However, it seems unlikely to me that users on small screens would intentionally click/tap the Sidebar (when not scrolling). I also think that they're unlikely to miss the links in the Sidebar while clicking/tapping.

I like that your current approach solves the more annoying issue of the Sidebar not closing when it should 😄 

>  What I could do is add it to the 'a' tags in the sidebar. What do you think?

This could work — I'm concerned about whether adding click-listeners to the `<a>` elements will override the browser-default behavior of scrolling and changing the URL hash. Give it a try, and let me know what you find!

**Comment by bjthompson805** [comment]
I tried to simply move the click listener from the `<ul>` to its descendant `<a>` nodes, but it didn't work, so I went with your idea.

I've pushed the changes, where I've moved the logic from useEffect() to an onClick on the `<nav>`.

**Review by seshrs — APPROVED** [review]
This looks great! Thanks again for proposing this fix 😃

**Pull request merged**